### PR TITLE
Add JIRA version creation as a build step.

### PR DESCRIPTION
**Pull request by MarcinCz**
Gives the possibility to create new JIRA version as a build step. With that user can create multiple versions in the single build process and can trigger creation conditionally.

---
This change is [Reviewable](https://reviewable.io/reviews/jenkinsci/jira-plugin/102)

### CODE REVIEW

File: src/main/java/hudson/plugins/jira/JiraVersionCreator.java [hunk 1]
```diff
@@ -1,30 +1,25 @@
...
-import static org.hamcrest.Matchers.equalTo;
 
 /**
  * A build step which creates new JIRA version
  *
  * @author Artem Koshelev artkoshelev@gmail.com
+ * @deprecated Replaced by {@link JiraVersionCreatorBuilder}. Read its description to see why.
+ * 	Kept for backward compatibility.
  */
 public class JiraVersionCreator extends Notifier {
```
Comment (warden) [inline src/main/java/hudson/plugins/jira/JiraVersionCreator.java]: Can you also add `@Deprecated` for the class?

**Comment by warden** [comment]
LGTM, @artkoshelev, @olamy any comments on this one?

File: src/main/java/hudson/plugins/jira/JiraVersionCreator.java [hunk 2]
```diff
@@ -36,7 +31,8 @@ public JiraVersionCreator(String jiraVersion, String jiraProjectKey) {
         this.jiraProjectKey = jiraProjectKey;
     }
 
-    public BuildStepMonitor getRequiredMonitorService() {
+    @Override
+	public BuildStepMonitor getRequiredMonitorService() {
```
Comment (artkoshelev) [inline src/main/java/hudson/plugins/jira/JiraVersionCreator.java]: looks like broken indentation

Comment (warden) [inline src/main/java/hudson/plugins/jira/JiraVersionCreator.java]: +1

**Pull request merged**